File: backend/app/schemas/suggestion.py
from pydantic import BaseModel
from typing import List, Optional

class SuggestionBase(BaseModel):
    title: str
    description: str

class SuggestionCreate(SuggestionBase):
    pass

class SuggestionResponse(SuggestionBase):
    id: int
    user_id: int
    likes_count: int
    dislikes_count: int
    user_has_liked: bool
    user_has_disliked: bool
    
    class Config:
        from_attributes = True

class SuggestionList(BaseModel):
    suggestions: List[SuggestionResponse]
    total: int

File: frontend/types/suggestion.ts
export interface Suggestion {
  id: number;
  title: string;
  description: string;
  user_id: number;
  user_name: string;
  likes_count: number;
  dislikes_count: number;
  user_has_liked: boolean;
  user_has_disliked: boolean;
}

export interface SuggestionList {
  suggestions: Suggestion[];
  total: number;
}

export interface CreateSuggestionData {
  title: string;
  description: string;
}